app/components/Navigation.tsx

```diff
@@ -0,0 +1,101 @@
+import { memo, useState, Fragment, useMemo } from "react";
+import Home from "~/components/Home";
+import Profile from "~/components/Profile";
+import { useAuth0 } from "@auth0/auth0-react";
+import { Menu, Transition } from '@headlessui/react'
+
+//TODO: fix style of profile picture (maybe expand the image when the menu is opened as well)
+
+const Navigation = memo(function Nav({ apiBaseURL }: { apiBaseURL: string }){
+    const [darkMode, setDarkMode] = useState(true);
+    const [pageChanged, setPageChanged] = useState(true);
+
+    const changePage = (item: number) => {
+        switch(item){
+            case 1:
+                setPageChanged(true);
+                break;
+            case 2:
+                setPageChanged(false);
+                break;
+            case 3:
+                setDarkMode(darkMode => !darkMode);
+                break;  
+            case 4:
+                isAuthenticated ? logout() : loginWithRedirect();
+                break;       
+        }
+    }
+
+    const { loginWithRedirect, logout, user, isAuthenticated, isLoading } = useAuth0();
+
+    const userProfile = {
+        name: user?.name ?? "John Doe",
+        email: user?.email ?? 'johndoe@example.com',
+        imageUrl: isAuthenticated ? user?.picture : 'https://www.vecteezy.com/vector-art/550980-user-icon-vector',
+    }
+
+    const userNavigation = [
+        { name: 'Home', page: 1 },
```

Comment:
Might it be clearer to, instead of setting a `page` property here, giving them all the same onclick which invokes `changePage` with the page number, and having a big switch statement there, to put an `onclick` property here set to a closure?